### ISSUE TITLE
sidenav closes on desktop despite sidenav-fixed class

### ISSUE THREAD

**Issue by JanMalch** [Dogfalo/materialize#5437]
The sidenav shouldn't be closing with a click on a `sidenav-close` element, when the sidenav has the `sidenav-fixed` class.
It works as expected on mobile but on desktop size, the sidenav disappears, even tho it's supposes to be fixed.

**Comment by Dogfalo** [comment]
This is the case in 1.0 alpha

**Comment by yannick9906** [comment]
Sorry to reopen this issue. But is there any reason why the sidenav should also close on a fixed sidenav on desktop sizes in 1.0 alpha? Is this intended to be like this?

**Comment by Dogfalo** [comment]
I don't quite understand what you mean. If possible, a codepen of the unintended behavior is helpful

**Comment by JanMalch** [comment]
I created a Codepen for this issue: https://codepen.io/anon/pen/rpwGag

**Comment by Dogfalo** [comment]
Yes this is intended. An example of this interaction is displayed on the inbox.google.com site

**Comment by yannick9906** [comment]
Uhm. I use Inbox every day. As it does not have a responsive design for mobile devices it does not need to automatically close the Sidenav. When I tested this the sidenav stays open even when I click on one of the links.

**Comment by JanMalch** [comment]
The sidenav indeed doesn't close on the desktop-view of inbox.google.com.
Also I don't quite understand what the point of the `sidenav-fixed` class is, if this behaviour is actually intended..

>Add the class `fixed` to have the sideNav be **fixed and open on large screens** and hides to the regular functionality on smaller screens.

**Comment by Dogfalo** [comment]
I see, I was talking originally about the hamburger icon still functioning as close even on desktop, but I see the problem you're talking about now.

**Comment by orsonok** [comment]
https://codepen.io/anon/pen/rpwGag
In your Codepen change:
var sideNavEl = document.getElementById("slide-out");
to
var sideNavEl = document.getElementById("#slide-out");

**Comment by JanMalch** [comment]
`getElementById` doesn't need a hashtag? It doesn't work that way at all.

**Comment by Dogfalo** [comment]
Fixed in 298a87f401dbbde5e48e2108e58a4feda2420ef4

**Comment by JanMalch** [comment]
Sorry to reopen this issue. I updated my [CodePen example](https://codepen.io/anon/pen/paGvmL) with the latest [version from cloudflare](https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0-alpha.4/js/materialize.min.js) but the sidenav still closes on desktop size.
I have the `sidenav-fixed` class on my `<ul>` element and the links have `class="sidenav-close"`. I initialize with `M.Sidenav.init(sideNavEl);` and the returned element actually has the property `isFixed` set to `true` but it still doesn't work on Desktop. Mobile behaviour is still as expected.
After clicking on a link on desktop, the `<ul>` element has `style="transform: translateX(-105%);"`.
Am I doing something wrong? Did the class names change or anything?

**Comment by Dahkon** [comment]
Same here with last version as of today.
I don't know if it might help but it's working fine when the links are differents from the current page (without reload, i'm on a single page app).

**Comment by JanMalch** [comment]
I'm also working in an Angular app.
I haven't tested it with a static website and external links.

**Comment by Dahkon** [comment]
A quick fix for me : 
```
SASS
	.sidenav {
		@media only screen and (min-width : 992px) {
			transform: translateX(0px) !important;
		}
        }
```

Another problem on Android when the keyboard shows up on a search field, the sidenav close.
I had to listen on focusin and out to add/remove a class on side nav : 
```
.fix-keyboard {
			transform: translateX(0px) !important;
		}
```

Hope it helps
